### PR TITLE
Add recents client

### PR DESCRIPTION
**Pull request by snappet-panksy**
This is a PR to add the client for getting the recents. I'm not too sure how you what you want for unit tests and how the the integration tests to be setup though. I can add unit tests to verify the RequestsCorrectURL but it also feels like I should add some for the converter. Let me know what you think, then I'll add more to this PR

### CODE REVIEW

**Comment by DavidRouyer** [comment]
@snappet-panksy Thanks

- Maybe you should use an enum for the **items** parameter, instead of doing a method for each value of items
- Unit tests check if the parameters are correctly passed to the Connection (you have a lot of examples in https://github.com/DavidRouyer/pipedrive-dotnet/tree/master/src/Pipedrive.net.Tests/Clients)
- Integration tests check if the methods are working :blush:. You just need to add your sandbox API token in the **PIPEDRIVE_APITOKEN** environment variable https://github.com/DavidRouyer/pipedrive-dotnet/blob/master/src/Pipedrive.net.Tests.Integration/Helper.cs#L11

**Comment by snappet-panksy** [comment]
Great thanks. Good idea on the Enum.

Do you have some kind of seed script or data import for the sandbox? I can see that when I run the integration tests on my Sandbox some data is missing.

Thanks

**Comment by DavidRouyer** [comment]
Integration tests should create and delete by itself the needed objects for the tests, but unfortunately it's not the case right now. I don't know how to deal elegantly with the API rate limit either.
You can create and test against the Recents client tho since they're indépendant.

**Comment by snappet-panksy** [comment]
I added the tests and integration tests so all ready for review now.

File: src/Pipedrive.net/IPipedriveClient.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public interface IPipedriveClient
         IPersonFieldsClient PersonField { get; }
 
         IPipelinesClient Pipeline { get; }
-
+        IRecentsClient Recents { get; }
```
Comment (DavidRouyer) [inline src/Pipedrive.net/IPipedriveClient.cs]: Please, add space before and after that line

File: src/Pipedrive.net/Clients/RecentsClient.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            var options = new ApiOptions
+            {
+                StartPage = filters.StartPage,
+                PageCount = filters.PageCount,
+                PageSize = filters.PageSize
+            };
+
+            return ApiConnection.GetAll<Recents>(ApiUrls.Recents(), parameters,options);
+        }
+        public Task<IReadOnlyList<Recents>> GetDealRecents(DateTime sinceWhen)
```
Comment (DavidRouyer) [inline src/Pipedrive.net/Clients/RecentsClient.cs]: Please, remove that method. `GetAll` handles all the call to the **recents** API.

File: src/Pipedrive.net/Clients/RecentsClient.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            var parameters = new Dictionary<string, string>
+            {
+                { "since_timestamp", sinceWhen.ToString("yyyy-MM-dd HH:mm:ss") },
+                { "items", "deal" }
+            };
+
+            return ApiConnection.GetAll<Recents>(ApiUrls.Recents(), parameters);
+        }
+        public Task<IReadOnlyList<Recents>> GetActivityRecents(DateTime sinceWhen)
```
Comment (DavidRouyer) [inline src/Pipedrive.net/Clients/RecentsClient.cs]: Please, remove that method. `GetAll` handles all the call to the **recents** API.

File: src/Pipedrive.net/Models/Response/SimpleDeal.cs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 namespace Pipedrive
 {
-    public class SimpleDeal
+    public class SimpleDeal : IRecentsEntity
```
Comment (DavidRouyer) [inline src/Pipedrive.net/Models/Response/SimpleDeal.cs]: Please, revert this change

File: src/Pipedrive.net/Models/Response/Recents.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Pipedrive
+{
+    [JsonConverter(typeof(RecentsConverter))]
+    public class Recents
+    {
+        public long Id { get; set; }
+
+        [JsonProperty("item")]
+        public string Object { get; set; }
```
Comment (DavidRouyer) [inline src/Pipedrive.net/Models/Response/Recents.cs]: Please, call this property `Item` to match the API
